Microsoft.Research/RegressionTest/ClousotTests/AsyncTestDriver.cs

```diff
@@ -20,147 +9,146 @@
+        public static readonly uint MaxWaitHandles_AllButOne = Math.Max(1, (uint)(Environment.ProcessorCount - 1));
 
-    private readonly Action<Options, Output> action;
-    private readonly IsolatedAction<Options> actionDelegate;
-    private readonly Func<Options, bool> skipTest;
-    private Dictionary<string, IAsyncResult> testAsyncResults;
-    private readonly uint maxWaitHandles;
-    private WaitHandle[] waitHandles;
-    private int nbWaitHandles;
-    private bool beginTestsProcessed = false;
-    private bool orderReversed = false;
+        private static readonly int SingleTestMaxWait = 200000;
 
-    public string BeginMessage;
+        private readonly Action<Options, Output> action;
+        private readonly IsolatedAction<Options> actionDelegate;
+        private readonly Func<Options, bool> skipTest;
+        private Dictionary<string, IAsyncResult> testAsyncResults;
+        private readonly uint maxWaitHandles;
+        private WaitHandle[] waitHandles;
+        private int nbWaitHandles;
+        private bool beginTestsProcessed = false;
+        private bool orderReversed = false;
 
-    public AsyncTestDriver(Action<Options, Output> action, Func<Options, bool> skipTest)
-      : this(action, skipTest, MaxWaitHandles_Default)
-    { }
+        public string BeginMessage;
 
-    public AsyncTestDriver(Action<Options, Output> action, Func<Options, bool> skipTest, uint maxWaitHandles)
-    {
-      this.action = action;
-      this.actionDelegate = this.ActionAsIsolated;
-      this.skipTest = skipTest;
-      this.maxWaitHandles = maxWaitHandles;
-    }
+        public AsyncTestDriver(Action<Options, Output> action, Func<Options, bool> skipTest)
+          : this(action, skipTest, MaxWaitHandles_Default)
+        { }
 
-    // We have no control on the order of the tests, so we make sure
-    // to always call Begin before End
-
-    public void BeginTest(Options options)
-    {
-      if (this.skipTest(options))
-        return;
+        public AsyncTestDriver(Action<Options, Output> action, Func<Options, bool> skipTest, uint maxWaitHandles)
+        {
+            this.action = action;
+            actionDelegate = this.ActionAsIsolated;
+            this.skipTest = skipTest;
+            this.maxWaitHandles = maxWaitHandles;
+        }
 
-      this.beginTestsProcessed = true;
+        // We have no control on the order of the tests, so we make sure
+        // to always call Begin before End
 
-      if (this.orderReversed)
-        this.EndTestInternal(options);
-      else
-        this.BeginTestInternal(options);
-    }
+        public void BeginTest(Options options)
+        {
+            if (skipTest(options))
+                return;
 
-    public void EndTest(Options options)
-    {
-      if (this.skipTest(options))
-        return;
+            beginTestsProcessed = true;
 
-      if (!this.beginTestsProcessed)
-        this.orderReversed = true;
+            if (orderReversed)
```

Comment:
StyleCopAnalyzers and StyleCop do not currently have a rule to enforce this, though one could certainly be proposed. However, there are a couple additional things to keep in mind:

1. I'm very much hoping that the Code Contracts projects can be moved from the Microsoft organization to the dotnet organization. If it does move here, it would make sense to use the formatting rules of other projects in that organization even if they deviate somewhat from our personal preferences.
1. The initial formatting is being done with dotnet/codeformatter (both in support of the first item and because it is compatible with Visual Studio's default formatting settings) so strict brace inclusion rules might be best saved for a later time.